src/sentry/models/actor.py

```diff
@@ -114,16 +115,18 @@ def get_actor_id_for_user(user: Union["User", RpcUser]):
         return user.actor_id
     # Temporary Dual write
     # Until we have indexes back online, we have to account for potential race condition on user creation
-    with transaction.atomic():
-        actors_for_user = Actor.objects.filter(type=ACTOR_TYPES["user"], user_id=user.id).all()
-        if len(actors_for_user) > 0:
-            actor = actors_for_user[0]
-        else:
-            actor = Actor.objects.create(type=ACTOR_TYPES["user"], user_id=user.id)
-        # Just clear other actors without allowing orm interaction, these won't be important after the following update.
-        Actor.objects.filter(type=ACTOR_TYPES["user"], user_id=user.id).exclude(id=actor.id).update(
-            user_id=None
-        )
+    lock = f"user-actor-lock:{user.id}"
+    with locks.get(lock, duration=60, name="actor-user-lock").blocking_acquire(0.020, 2):
```

Comment:
That will likely be fine because the logic is consistently short and simple, and most users already have a single backfilled value right now.

Q though: why not, just use select for update then?  Because atleast then the lock is consistent with the database transaction state.

But a very minor nit, I suspect this is fine.

Reply:
Could've, this was already written and tested and I wasn't exactly sure about the failure modes of conflicting transactions, so just synchronizing it seemed fine.

Reply:
I agree.  This makes sense for effort and outcome.